homeassistant/components/doods/image_processing.py

```diff
@@ -0,0 +1,363 @@
+
+    @property
+    def name(self):
+        """Return the name of the image processor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the entity."""
+        return self._total_matches
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        return {
+            ATTR_MATCHES: self._matches,
+            ATTR_SUMMARY: {
+                label: len(values) for label, values in self._matches.items()
+            },
+            ATTR_TOTAL_MATCHES: self._total_matches,
+        }
+
+    def _save_image(self, image, matches, paths):
+        img = Image.open(io.BytesIO(bytearray(image))).convert("RGB")
+        img_width, img_height = img.size
+        draw = ImageDraw.Draw(img)
+
+        # Draw custom global region/area
+        if self._area != [0, 0, 1, 1]:
+            draw_box(
+                draw, self._area, img_width, img_height, "Detection Area", (0, 255, 255)
+            )
+
+        for label, values in matches.items():
+
+            # Draw custom label regions/areas
+            if label in self._label_areas and self._label_areas[label] != [0, 0, 1, 1]:
+                box_label = "{} Detection Area".format(label.capitalize())
+                draw_box(
+                    draw,
+                    self._label_areas[label],
+                    img_width,
+                    img_height,
+                    box_label,
+                    (0, 255, 0),
+                )
+
+            # Draw detected objects
+            for instance in values:
+                box_label = "{0} {1:.1f}%".format(label, instance["score"])
+                # Already scaled, use 1 for width and height
+                draw_box(
+                    draw,
+                    instance["box"],
+                    img_width,
+                    img_height,
+                    box_label,
+                    (255, 255, 0),
+                )
+
+        for path in paths:
+            _LOGGER.info("Saving results image to %s", path)
+            img.save(path)
+
+    def process_image(self, image):
+        """Process the image."""
+
+        from PIL import Image
+        import io
+
+        img = Image.open(io.BytesIO(bytearray(image)))
+        img_width, img_height = img.size
+
+        if self._aspect and abs((img_width / img_height) - self._aspect) > 0.1:
+            _LOGGER.debug(
+                "The image aspect: %s and the detector aspect: %s differ by more than 0.1",
+                (img_width / img_height),
+                self._aspect,
+            )
+
+        # Run detection
+        start = time.time()
+        response = self._doods.detect(image, self._dconfig)
+        _LOGGER.info(
+            "doods detect: %s response: %s duration: %s",
+            self._dconfig,
+            response,
+            time.time() - start,
+        )
+
+        matches = {}
+        total_matches = 0
+
+        if response:
```

Comment:
```suggestion
        if not response or "error" in response:
            if "error" in response:
                _LOGGER.error(response["error"])
            self._matches = matches
            self._total_matches = total_matches
            return
        for detection in response["detections"]:
```

Reply:
Actually, I found a bug that I fixed with file output that I think will make the linter happy anyways

Reply:
Please change according to above too.

Reply:
I did it by hand...